src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -309,7 +308,6 @@ public void checkAndAuthenticateRequest(SecurityRequestChannel requestChannel) t
             if (!registry.authenticate(requestChannel)) {
                 // another roundtrip
                 org.apache.logging.log4j.ThreadContext.remove("user");
-                return;
```

Comment:
`return` is unnecessary as the last statement in a 'void' method

Reply:
While reading the security code, I cleaned up some unnecessary parts. This is not a major PR, so please review it lightly.